### PR TITLE
Move substituteParams into its own module

### PR DESCRIPTION
**Pull request by valadaptive**
This is a first step towards a macro system refactor.

I've decided not to import name1 and name2 directly into the new module, but instead pass them in to all substituteParams calls.

I have no way of knowing whether name1 and name2 are actually useful in any given context where substituteParams is called, but this leaves the door open to future refactors without breaking anything, and by making it explicit, this should help people think about whether they *really* want to pass name1 and name2 in.

If you're up to it, it may be worth going through some of the substituteParams calls now, and removing name1 and name2 from those which don't need them.

### CODE REVIEW

**Comment by Cohee1207** [comment]
This will break downloadable extensions and third-party extensions that we're not aware of.
I suggest that you leave the old substituteParams function as a proxy and wrap the call into a new function that will accept name1, name2, and other parameters, maybe even as an object to simplify future additions and refactorings.

**Comment by valadaptive** [comment]
That approach is actually way nicer--I'll be able to turn `substituteParams` into a compatibility shim for the new macro evaluation API in whatever form it takes.

File: public/script.js [hunk 1]
```diff
@@ -2146,187 +2061,9 @@ function getCurrentSwipeId() {
...
-    content = content.replace(/{{idle_duration}}/gi, () => getTimeSinceLastMessage());
-    content = content.replace(/{{time_UTC([-+]\d+)}}/gi, (_, offset) => {
-        const utcOffset = parseInt(offset, 10);
-        const utcTime = moment().utc().utcOffset(utcOffset).format('LT');
-        return utcTime;
-    });
-    content = bannedWordsReplace(content);
-    content = randomReplace(content);
-    return content;
+    return evaluateMacros(content, _name1 ?? name1, _name2 ?? name2, _original, _group ?? name2);
```
Comment (Cohee1207) [inline public/script.js]: `_replaceCharacterCard` param is not passed down to the shim. Is that correct?

Comment (valadaptive) [reply public/script.js]: Good catch--I missed it because it's not in the JSDoc

**Pull request merged**